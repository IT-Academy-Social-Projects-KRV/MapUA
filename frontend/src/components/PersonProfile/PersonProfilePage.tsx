// eslint-disable-next-line no-unused-vars
import React, { useEffect, useState } from 'react';
import { Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { useTypedSelector } from 'redux/hooks/useTypedSelector';
import { useParams } from 'react-router-dom';
import { useTypedDispatch } from 'redux/hooks/useTypedDispatch';
import userImageNotFound from '../../static/user-image-not-found.png';
import {
  ProfileAvatar,
  ProfileContentWrapper,
  ProfileFormWrapper,
  ProfileUserWrapper,
  SubsrcibeButton
} from '../design/StyledProfile';
import BasicTabs from './BasicTabs';

export default function PersonProfilePage() {
  const { t } = useTranslation();
  const {
    loading: userLoading,
    data: { _id: otherUserId, displayName, imageUrl: userAvatar }
  } = useTypedSelector(state => state.otherUserData);
  const {
    data: { _id: userId, subscriptions }
  } = useTypedSelector(state => state.userData);

  const { data: isAuthorized } = useTypedSelector(
    state => state.isUserAuthorized
  );

  const { fetchOtherUserData, toogleUserSubscription } = useTypedDispatch();

  const isSubscribed = subscriptions.includes(otherUserId);

  const params = useParams();
  useEffect(() => {
    fetchOtherUserData(params?.id || '');
  }, [params.id]);

  if (userLoading) {
    return <h1>{t('profile.profile.loading')}</h1>;
  }

  const handleSubscription = () => {
    if (isAuthorized) {
      toogleUserSubscription(
        otherUserId,
        userId,
        t('profile.profilePage.profilePageUpdatedSuccessfully')
      );
    }
  };
  return (
    <ProfileFormWrapper>
      <ProfileContentWrapper sx={{ height: 'auto' }}>
        <ProfileAvatar
          aria-label="avatar"
          src={userAvatar || userImageNotFound}
        />
        <Typography mt={2} variant="h5" component="h4" align="center">
          {displayName === undefined
            ? `${t('profile.profilePage.yourName')}`
            : displayName}
        </Typography>
        {isAuthorized.isAuthorized && (
          <SubsrcibeButton
            size="large"
            variant="contained"
            onClick={handleSubscription}
          >
            {!isSubscribed
              ? t('profile.profilePage.subscribe')
              : t('profile.profilePage.unsubscribe')}
          </SubsrcibeButton>
        )}
      </ProfileContentWrapper>
      <ProfileUserWrapper>
        <BasicTabs />
      </ProfileUserWrapper>
    </ProfileFormWrapper>
  );
}
