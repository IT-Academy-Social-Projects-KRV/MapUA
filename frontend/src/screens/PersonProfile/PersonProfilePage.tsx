// eslint-disable-next-line no-unused-vars
import React, { useEffect } from 'react';
import { Button, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import {
  selectOtherUserAvatar,
  selectOtherUserDataLoading,
  selectOtherUserDisplayName,
  selectOtherUserId,
  selectOtherUserRole
} from 'redux/memoizedSelectors/otherUserDataSelectors';
import { selectUserDataSubscriptions } from 'redux/memoizedSelectors/userDataSelectors';
import {
  selectIsUserAuthorized,
  selectUserRole
} from 'redux/memoizedSelectors/isUserAuthorizedSelectors';
import { useTypedSelector } from 'redux/hooks/useTypedSelector';
import { useParams } from 'react-router-dom';
import { useTypedDispatch } from 'redux/hooks/useTypedDispatch';
import userImageNotFound from '../../static/user-image-not-found.png';
import {
  ProfileAvatar,
  ProfileContentWrapper,
  ProfileFormWrapper,
  ProfileUserWrapper,
  SubsrcibeButton
} from '../../components/design/StyledProfile';
import BasicTabs from './BasicTabs';
import CircularLoader from '../../components/CircularLoader/CircularLoader';

export default function PersonProfilePage() {
  const { t } = useTranslation();

  const userLoading = useTypedSelector(selectOtherUserDataLoading);
  const otherUserId = useTypedSelector(selectOtherUserId);
  const displayName = useTypedSelector(selectOtherUserDisplayName);
  const userAvatar = useTypedSelector(selectOtherUserAvatar);
  const otherUserRole = useTypedSelector(selectOtherUserRole);
  const subscriptions = useTypedSelector(selectUserDataSubscriptions);
  const isAuthorized = useTypedSelector(selectIsUserAuthorized);
  const myRole = useTypedSelector(selectUserRole);

  const {
    fetchOtherUserData,
    toggleUserSubscription,
    toggleUserBan,
    toggleModeratorRights
  } = useTypedDispatch();

  const isSubscribed = subscriptions.some((s: any) => s._id === otherUserId);

  const params = useParams();
  useEffect(() => {
    fetchOtherUserData(params?.id || '');
  }, [params.id]);

  if (userLoading) {
    return <CircularLoader />;
  }

  const handleModeratorRights = () => {
    if (myRole === 'admin') {
      toggleModeratorRights(
        otherUserId,
        t('personalProfile.personalProfilePage.rightsUpdated')
      );
    }
  };

  const handleSubscription = () => {
    if (isAuthorized) {
      toggleUserSubscription(
        otherUserId,
        t('personalProfile.personalProfilePage.subscriptionUpdated')
      );
    }
  };

  const toogleBanStatus = (): string => {
    if (otherUserRole === 'user') {
      return 'bannedUser';
    }
    return 'user';
  };

  const handleBan = () => {
    if (isAuthorized) {
      toggleUserBan(
        otherUserId,
        toogleBanStatus(),
        t('personalProfile.personalProfilePage.banningUpdated')
      );
    }
  };

  return (
    <ProfileFormWrapper>
      <ProfileContentWrapper sx={{ height: 'auto' }}>
        <ProfileAvatar
          aria-label="avatar"
          src={userAvatar || userImageNotFound}
        />
        <Typography mt={2} variant="h5" component="h4" align="center">
          {!displayName ? `${t('profile.profilePage.yourName')}` : displayName}
        </Typography>

        {isAuthorized && (
          <>
            <SubsrcibeButton
              size="large"
              variant="contained"
              onClick={handleSubscription}
            >
              {isSubscribed
                ? t('profile.profilePage.unsubscribe')
                : t('profile.profilePage.subscribe')}
            </SubsrcibeButton>

            {otherUserRole !== 'admin' &&
              otherUserRole !== 'moderator' &&
              (myRole === 'moderator' || myRole === 'admin') && (
                <SubsrcibeButton
                  sx={{ mb: '40px' }}
                  size="large"
                  variant="contained"
                  onClick={handleBan}
                >
                  {otherUserRole === 'bannedUser'
                    ? t('profile.profilePage.unban')
                    : t('profile.profilePage.ban')}
                </SubsrcibeButton>
              )}

            {myRole === 'admin' && otherUserRole !== 'admin' && (
              <Button variant="outlined" onClick={handleModeratorRights}>
                {otherUserRole === 'moderator'
                  ? t('profile.profilePage.removeModerator')
                  : t('profile.profilePage.createModerator')}
              </Button>
            )}
          </>
        )}
      </ProfileContentWrapper>
      <ProfileUserWrapper>
        <BasicTabs />
      </ProfileUserWrapper>
    </ProfileFormWrapper>
  );
}
