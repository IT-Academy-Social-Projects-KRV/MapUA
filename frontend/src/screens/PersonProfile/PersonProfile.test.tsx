/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/react-in-jsx-scope */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PersonProfilePage from './PersonProfilePage';

describe('PersonProfilePage', () => {
  const initialState = {
    otherUserData: {
      loading: false,
      error: null,
      success: false,
      data: {
        _id: '',
        displayName: '',
        description: '',
        imageUrl: '',
        role: '',
        subscribers: [],
        subscriptions: [1, 2, 3],
        favorite: [],
        visited: [],
        personalLocations: []
      }
    },
    userData: {
      loading: false,
      error: null,
      success: false,
      data: []
    },
    isUserAuthorized: {
      loading: false,
      error: null,
      success: false,
      data: []
    }
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  it('should render TopLocations', () => {
    const store = mockStore(initialState);
    const component = render(
      <Provider store={store}>
        <Router>
          <PersonProfilePage />
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
