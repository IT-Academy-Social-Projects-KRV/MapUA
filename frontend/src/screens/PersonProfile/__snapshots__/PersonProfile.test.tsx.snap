// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonProfilePage should render TopLocations 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-l1w0ah"
      >
        <div
          class="MuiBox-root css-9ls6wk"
        >
          <div
            aria-label="avatar"
            class="MuiAvatar-root MuiAvatar-circular css-15o4hbz-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="user-image-not-found.png"
            />
          </div>
          <h4
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1sly7xo-MuiTypography-root"
          />
        </div>
        <div
          class="MuiBox-root css-rxzivr"
        >
          <div
            class="MuiBox-root css-8atqhb"
          >
            <div
              class="MuiBox-root css-1gsv261"
            >
              <div
                class="MuiTabs-root css-1ujnqem-MuiTabs-root"
              >
                <div
                  class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                  style="overflow: hidden; margin-bottom: 0px;"
                >
                  <div
                    aria-label="basic tabs example"
                    class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                    role="tablist"
                  >
                    <button
                      aria-controls="simple-tabpanel-0"
                      aria-selected="true"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-0"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      profile.basicTabs.description
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-controls="simple-tabpanel-1"
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-1"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      profile.basicTabs.subscribers
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-controls="simple-tabpanel-2"
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-2"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      profile.basicTabs.subscriptions
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <span
                    class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                    style="left: 0px; width: 0px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                aria-labelledby="simple-tab-0"
                class="MuiBox-root css-1dgopwc"
                id="simple-tabpanel-0"
                role="tabpanel"
              >
                <div
                  class="MuiBox-root css-19kzrtu"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      profile.basicTabs.noDescription
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                aria-labelledby="simple-tab-1"
                class="MuiBox-root css-1dgopwc"
                hidden=""
                id="simple-tabpanel-1"
                role="tabpanel"
              />
            </div>
            <div
              aria-labelledby="simple-tab-2"
              class="MuiBox-root css-1dgopwc"
              hidden=""
              id="simple-tabpanel-2"
              role="tabpanel"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-l1w0ah"
    >
      <div
        class="MuiBox-root css-9ls6wk"
      >
        <div
          aria-label="avatar"
          class="MuiAvatar-root MuiAvatar-circular css-15o4hbz-MuiAvatar-root"
        >
          <img
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="user-image-not-found.png"
          />
        </div>
        <h4
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1sly7xo-MuiTypography-root"
        />
      </div>
      <div
        class="MuiBox-root css-rxzivr"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="MuiBox-root css-1gsv261"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="basic tabs example"
                  class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="simple-tabpanel-0"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-0"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    profile.basicTabs.description
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-1"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-1"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    profile.basicTabs.subscribers
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-2"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-2"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    profile.basicTabs.subscriptions
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              aria-labelledby="simple-tab-0"
              class="MuiBox-root css-1dgopwc"
              id="simple-tabpanel-0"
              role="tabpanel"
            >
              <div
                class="MuiBox-root css-19kzrtu"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    profile.basicTabs.noDescription
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              aria-labelledby="simple-tab-1"
              class="MuiBox-root css-1dgopwc"
              hidden=""
              id="simple-tabpanel-1"
              role="tabpanel"
            />
          </div>
          <div
            aria-labelledby="simple-tab-2"
            class="MuiBox-root css-1dgopwc"
            hidden=""
            id="simple-tabpanel-2"
            role="tabpanel"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TopLocations should render TopLocations 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-l1w0ah"
        >
          <div
            class="MuiBox-root css-9ls6wk"
          >
            <div
              aria-label="avatar"
              class="MuiAvatar-root MuiAvatar-circular css-15o4hbz-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="user-image-not-found.png"
              />
            </div>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1sly7xo-MuiTypography-root"
            />
          </div>
          <div
            class="MuiBox-root css-rxzivr"
          >
            <div
              class="MuiBox-root css-8atqhb"
            >
              <div
                class="MuiBox-root css-1gsv261"
              >
                <div
                  class="MuiTabs-root css-1ujnqem-MuiTabs-root"
                >
                  <div
                    class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                    style="overflow: hidden; margin-bottom: 0px;"
                  >
                    <div
                      aria-label="basic tabs example"
                      class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                      role="tablist"
                    >
                      <button
                        aria-controls="simple-tabpanel-0"
                        aria-selected="true"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                        id="simple-tab-0"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        profile.basicTabs.description
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-controls="simple-tabpanel-1"
                        aria-selected="false"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                        id="simple-tab-1"
                        role="tab"
                        tabindex="-1"
                        type="button"
                      >
                        profile.basicTabs.subscribers
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        aria-controls="simple-tabpanel-2"
                        aria-selected="false"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                        id="simple-tab-2"
                        role="tab"
                        tabindex="-1"
                        type="button"
                      >
                        profile.basicTabs.subscriptions
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <span
                      class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                      style="left: 0px; width: 0px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  aria-labelledby="simple-tab-0"
                  class="MuiBox-root css-1dgopwc"
                  id="simple-tabpanel-0"
                  role="tabpanel"
                >
                  <div
                    class="MuiBox-root css-19kzrtu"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      >
                        profile.basicTabs.noDescription
                      </span>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  aria-labelledby="simple-tab-1"
                  class="MuiBox-root css-1dgopwc"
                  hidden=""
                  id="simple-tabpanel-1"
                  role="tabpanel"
                />
              </div>
              <div
                aria-labelledby="simple-tab-2"
                class="MuiBox-root css-1dgopwc"
                hidden=""
                id="simple-tabpanel-2"
                role="tabpanel"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-l1w0ah"
      >
        <div
          class="MuiBox-root css-9ls6wk"
        >
          <div
            aria-label="avatar"
            class="MuiAvatar-root MuiAvatar-circular css-15o4hbz-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="user-image-not-found.png"
            />
          </div>
          <h4
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1sly7xo-MuiTypography-root"
          />
        </div>
        <div
          class="MuiBox-root css-rxzivr"
        >
          <div
            class="MuiBox-root css-8atqhb"
          >
            <div
              class="MuiBox-root css-1gsv261"
            >
              <div
                class="MuiTabs-root css-1ujnqem-MuiTabs-root"
              >
                <div
                  class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                  style="overflow: hidden; margin-bottom: 0px;"
                >
                  <div
                    aria-label="basic tabs example"
                    class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                    role="tablist"
                  >
                    <button
                      aria-controls="simple-tabpanel-0"
                      aria-selected="true"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-0"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      profile.basicTabs.description
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-controls="simple-tabpanel-1"
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-1"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      profile.basicTabs.subscribers
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-controls="simple-tabpanel-2"
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-2"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      profile.basicTabs.subscriptions
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <span
                    class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                    style="left: 0px; width: 0px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                aria-labelledby="simple-tab-0"
                class="MuiBox-root css-1dgopwc"
                id="simple-tabpanel-0"
                role="tabpanel"
              >
                <div
                  class="MuiBox-root css-19kzrtu"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      profile.basicTabs.noDescription
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                aria-labelledby="simple-tab-1"
                class="MuiBox-root css-1dgopwc"
                hidden=""
                id="simple-tabpanel-1"
                role="tabpanel"
              />
            </div>
            <div
              aria-labelledby="simple-tab-2"
              class="MuiBox-root css-1dgopwc"
              hidden=""
              id="simple-tabpanel-2"
              role="tabpanel"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
